### PR TITLE
Added logs for notifications and activity start/finish

### PR DESCRIPTION
**Pull request by vmkhitaryanscn**

### CODE REVIEW

**Review by anq83 — CHANGES_REQUESTED** [review]
Follow the same logs format, add necessary id, names

File: src/entities/applet/model/hooks/useStartEntity.ts [hunk 1]
```diff
@@ -151,12 +153,23 @@ function useStartEntity({
           onRestart: () => {
             cleanUpMediaFiles({ activityId, appletId, eventId, order: 0 });
 
+            logger.log(
+              `[useStartEntity.startActivity]: Activity "${entityName}|${activityId}" restarted`,
```
Comment (anq83) [inline src/entities/applet/model/hooks/useStartEntity.ts:157]: Missed applet information

File: src/entities/applet/model/hooks/useStartEntity.ts [hunk 2]
```diff
@@ -151,12 +153,23 @@ function useStartEntity({
...
+            );
             activityStarted(appletId, activityId, eventId);
             resolve({ startedFromScratch: true });
           },
-          onResume: () => resolve({ startedFromScratch: false }),
+          onResume: () => {
+            logger.log(
+              `[useStartEntity.startActivity]: Activity "${entityName}|${activityId}" resumed`,
```
Comment (anq83) [inline src/entities/applet/model/hooks/useStartEntity.ts:164]: Missed applet information

File: src/entities/applet/model/hooks/useStartEntity.ts [hunk 3]
```diff
@@ -151,12 +153,23 @@ function useStartEntity({
...
+            );
+            return resolve({ startedFromScratch: false });
+          },
         });
       } else {
+        logger.log(
+          `[useStartEntity.startActivity]: Activity "${entityName}|${activityId}" started`,
```
Comment (anq83) [inline src/entities/applet/model/hooks/useStartEntity.ts:171]: Missed applet information

File: src/entities/applet/model/hooks/useStartEntity.ts [hunk 4]
```diff
@@ -250,18 +263,30 @@ function useStartEntity({
                 order: i,
               });
             }
+            logger.log(
+              `[useStartEntity.startFlow]: Flow "${entityName}|${flowId}" restarted`,
```
Comment (anq83) [inline src/entities/applet/model/hooks/useStartEntity.ts:267]: Missed applet information

File: src/entities/applet/model/hooks/useStartEntity.ts [hunk 5]
```diff
@@ -250,18 +263,30 @@ function useStartEntity({
...
             flowStarted(appletId, flowId, firstActivityId, eventId, 0);
             resolve({
               startedFromScratch: true,
             });
           },
-          onResume: () =>
-            resolve({
+          onResume: () => {
+            logger.log(
+              `[useStartEntity.startFlow]: Flow "${entityName}|${flowId}" resumed`,
```
Comment (anq83) [inline src/entities/applet/model/hooks/useStartEntity.ts:277]: Missed applet information

File: src/entities/applet/model/hooks/useStartEntity.ts [hunk 6]
```diff
@@ -250,18 +263,30 @@ function useStartEntity({
...
+
+            return resolve({
               startedFromScratch: false,
-            }),
+            });
+          },
         });
       } else {
+        logger.log(
+          `[useStartEntity.startFlow]: Flow "${entityName}|${flowId}" started`,
```
Comment (anq83) [inline src/entities/applet/model/hooks/useStartEntity.ts:287]: Missed applet information

File: src/widgets/survey/ui/Finish.tsx [hunk 7]
```diff
@@ -145,6 +150,14 @@ function FinishItem({
 
     const executionGroupKey = getExecutionGroupKey(progressRecord);
 
+    const logActivityName = getActivityName(activityId);
+
+    Logger.log(
```
Comment (anq83) [inline src/widgets/survey/ui/Finish.tsx:155]: Missed applet information

File: src/widgets/survey/ui/Intermediate.tsx [hunk 8]
```diff
@@ -206,6 +207,14 @@ function Intermediate({
       activityStorageRecord.answers,
     );
 
+    const logActivityName = getActivityName(activityId);
+
+    Logger.log(
```
Comment (anq83) [inline src/widgets/survey/ui/Intermediate.tsx:212]: Missed applet information

**Review by anq83 — CHANGES_REQUESTED** [review]

File: src/widgets/survey/ui/Finish.tsx [hunk 9]
```diff
@@ -145,6 +150,14 @@ function FinishItem({
...
+      `[Finish.completeActivity]: ${
+        flowId ? 'Flow' : 'Activity'
+      } ${logActivityName}|${activityId} completed`,
```
Comment (anq83) [inline src/widgets/survey/ui/Finish.tsx:158]: It seems the message is incorrect. We'd get: Flow [activity info] completed. If you're talking about flow then should be: Activity [activity info] within Flow [flow info] completed

Comment (anq83) [reply src/widgets/survey/ui/Finish.tsx:158]: And the next message - Flow [flow info] completed

Comment (anq83) [reply src/widgets/survey/ui/Finish.tsx:158]: Consider this for Intermediate screen as well

File: src/widgets/survey/ui/Intermediate.tsx [hunk 10]
```diff
@@ -206,6 +207,14 @@ function Intermediate({
...
+      `[Intermediate.completeActivity]: ${
+        flowId ? 'flow' : 'activity'
+      } ${logActivityName}|${activityId} completed`,
```
Comment (anq83) [inline src/widgets/survey/ui/Intermediate.tsx:215]: Message is wrong. See comment for Finish screen

**Review by anq83 — CHANGES_REQUESTED** [review]

File: src/entities/applet/model/hooks/useStartEntity.ts [hunk 11]
```diff
@@ -151,12 +153,23 @@ function useStartEntity({
...
-          onResume: () => resolve({ startedFromScratch: false }),
+          onResume: () => {
+            logger.log(
+              `[useStartEntity.startActivity]: Activity "${entityName}|${activityId}" resumed`,
+            );
+            return resolve({ startedFromScratch: false });
+          },
         });
       } else {
+        logger.log(
```
Comment (anq83) [inline src/entities/applet/model/hooks/useStartEntity.ts:170]: Please clarify to Veronika, aren't these new logs sensitive user data like user actions (user events) that collected during activity execution?

**Review by anq83 — CHANGES_REQUESTED** [review]

File: src/shared/lib/alerts/permissionAlerts.ts [hunk 12]
```diff
@@ -43,10 +51,18 @@ export const onAlarmPermissionsDisabled = () => {
       {
         text: t('permissions:alert_button_cancel'),
         style: 'cancel',
+        onPress: () => {
+          Logger.log(
+            '[permissionAlerts.onAlarmPermissionsDisabled] result: dismissed',
```
Comment (anq83) [inline src/shared/lib/alerts/permissionAlerts.ts:56]: Is such data acceptable for write into logs?

File: src/widgets/survey/ui/Finish.tsx [hunk 13]
```diff
@@ -145,6 +150,20 @@ function FinishItem({
 
     const executionGroupKey = getExecutionGroupKey(progressRecord);
 
+    const logActivityName = getActivityName(activityId);
+
+    const appletName = applet?.displayName;
+
+    Logger.log(
+      `[Finish.completeActivity]: Applet "${appletName}|${appletId}", Activity "${logActivityName}|${activityId}" completed`,
```
Comment (anq83) [inline src/widgets/survey/ui/Finish.tsx:158]: re-phrase, please

File: src/widgets/survey/ui/Intermediate.tsx [hunk 14]
```diff
@@ -155,6 +158,19 @@ function Intermediate({
       return;
     }
 
+    const appletName = applet?.displayName;
+    const currentActivityName = getActivityName(activityId);
+
+    if (flowId) {
```
Comment (anq83) [inline src/widgets/survey/ui/Intermediate.tsx:164]: flowId is always true in scope of Intermediate screen

File: src/widgets/survey/ui/Intermediate.tsx [hunk 15]
```diff
@@ -206,6 +233,20 @@ function Intermediate({
       activityStorageRecord.answers,
     );
 
+    const logActivityName = getActivityName(activityId);
+
+    const appletName = applet?.displayName;
+
+    if (flowId) {
```
Comment (anq83) [inline src/widgets/survey/ui/Intermediate.tsx:240]: The same: flowId is always true here

**Review by anq83 — CHANGES_REQUESTED** [review]

File: src/widgets/survey/ui/Intermediate.tsx [hunk 16]
```diff
@@ -155,6 +158,13 @@ function Intermediate({
       return;
     }
 
+    const appletName = applet?.displayName;
+    const currentActivityName = getActivityName(activityId);
+
+    Logger.log(
```
Comment (anq83) [inline src/widgets/survey/ui/Intermediate.tsx:164]: please, use nextActivity?.name / id

**Review by anq83 — APPROVED** [review]

**Review by anq83 — CHANGES_REQUESTED** [review]
one minor fix to do

**Review by BamMironov — APPROVED** [review]

**Review by anq83 — APPROVED** [review]

File: src/shared/lib/services/Logger.ts [hunk 17]
```diff
@@ -258,6 +306,8 @@ class Logger implements ILogger {
     }
 
     try {
+      await this.onBeforeSendLogs();
```
Comment (anq83) [inline src/shared/lib/services/Logger.ts:309]: Please move it below setBusy()

File: src/shared/lib/services/Logger.ts [hunk 18]
```diff
@@ -101,6 +103,52 @@ class Logger implements ILogger {
...
+    const {
+      getBrand,
+      getReadableVersion,
+      getBuildNumber,
+      getFirstInstallTime,
+      getFreeDiskStorage,
+      getLastUpdateTime,
+    } = DeviceInfo;
+
+    const deviceInfo = {
```
Comment (BamMironov) [inline src/shared/lib/services/Logger.ts:126]: Please add a type here

**Review by anq83 — DISMISSED** [review]

**Review by BamMironov — APPROVED** [review]

**Review by mbarsukou — APPROVED** [review]

**Review by mbarsukou — APPROVED** [review]

**Review by mbarsukou — APPROVED** [review]

**Pull request merged**